src/helpers/pivot/pivot_time_adapter.ts

```
@@ -1,10 +1,10 @@
-import { toNumber } from "../../functions/helpers";
+import { toJsDate, toNumber } from "../../functions/helpers";
```

I'm not sure this really helps, we are even more inconsistent than before with this. We can use dates for months, but not for quarters/years/weeks ? 

- for quarters it somehow works, but the autofill give us a big traceback `=PIVOT.HEADER(1,"create_date:year",2024,"create_date:quarter", DATE(2024,4,1))`
- it doesn't work for years `=PIVOT.HEADER(1,"create_date:year",DATE(2024, 12, 8))`
- it doesn't work for weeks `=PIVOT.HEADER(1,"create_date:week",DATE(2024,12,1))` give me `W45627 NaN`

I'm not sure the PR really helps if we don't accept dates everywhere. It will just lead to people trying to use dates for year/quarters/weeks and creating tickets when it crashes. Or at the very least we should have real errors in the cells that explain to the user what he did wrong, rather than trying to convert the input, returning a nonsensical value, and crashing at autofill.

Or maybe i'm wrong and they'll happily use `DATE` for days/months and nothing else 🤷

---

> we are even more inconsistent than before with this. We can use dates for months, but not for quarters/years/weeks ?

it's not more inconsistent with this PR. It already works now in master, I just change the default value in formulas.

> it doesn't work for years

it's much more readable and straight forward to have the year as a number `=PIVOT.HEADER(1,"create_date:year",2024)`. We could do something like `if (year > 3000) { it's probably a date }`, 

> it doesn't work for weeks

Group by week is an illusion. It doesn't exist :ghost:
(more seriously, we can't make it work client-side because only the server is able to convert date -> week number according to the user's locale)

> but the autofill give us a big traceback

pivot autofill is also an illusion. It doesn't exist :ghost: (I'll have a look)

---

> I'm not sure the PR really helps

the current situation (encouraging to use strings) is even worse